src/components/CuratorSelector.tsx

```diff
@@ -0,0 +1,92 @@
+import { Autocomplete, Avatar, Box, Typography } from '@mui/material';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { useQuery } from '@tanstack/react-query';
+import { fetchUsers } from '../api/roleApi';
+import { TicketSearchParam } from '../api/searchApi';
+import { RootState } from '../redux/store';
+import { InstitutionUser, RoleName } from '../types/user.types';
+import { dataTestId } from '../utils/dataTestIds';
+import { getInitials } from '../utils/general-helpers';
+import { getFullName } from '../utils/user-helpers';
+import { AutocompleteTextField } from './AutocompleteTextField';
+
+interface CuratorSelectorProps {
+  roleFilter: RoleName[];
+}
+
+export const CuratorSelector = ({ roleFilter }: CuratorSelectorProps) => {
+  const { t } = useTranslation();
+  const user = useSelector((store: RootState) => store.user);
+  const customerId = user?.customerId ?? '';
+
+  const history = useHistory();
+  const searchParams = new URLSearchParams(history.location.search);
+
+  const curatorsQuery = useQuery({
+    queryKey: ['curators', customerId, roleFilter],
+    enabled: !!customerId,
+    queryFn: () => fetchUsers(customerId, roleFilter),
+    meta: { errorMessage: t('feedback.error.get_curators_for_institution') },
+  });
+
+  // Ensure current user is first in list of curators
+  const curatorOptions = (curatorsQuery.data ?? []).sort((a, b) =>
+    a.username === user?.nvaUsername ? -1 : b.username === user?.nvaUsername ? 1 : 0
+  );
+
+  const selectedCurator =
+    curatorOptions.find((curator) => curator.username === searchParams.get(TicketSearchParam.Assignee)) ?? null;
+
+  const CuratorAvatar = (curator: InstitutionUser | null) => {
+    return (
+      <Avatar sx={{ height: '1.5rem', width: '1.5rem', fontSize: 'inherit', bgcolor: 'primary.main' }}>
+        {getInitials(getFullName(curator?.givenName, curator?.familyName))}
+      </Avatar>
+    );
+  };
```

Comment:
Syns denne er litt funky, men klarer ikke komme med noen god grunn til å gjøre det på noen annen måte heller :D 

Det er bare en funksjon som returnerer en `<Avatar>`, så min første tanke var om dette ikke egentlig er en egen komponent, som tar inn `curator` som prop. Det virker litt mer naturlig på et vis(?)

Så tenkte jeg på at det egentlig bare er en Avatar med noe default styling, så kanskje `StyledAvatar` er det vi egetnlig er ute ette?
```
const StyledAvatar = styled(Avatar)(({ theme }) => ({
  height: '1.5rem',
  width: '1.5rem',
  fontSize: 'inherit',
  bgcolor: theme.palette.primary.main,
}));

<StyledAvatar>{getInitials(getFullName(curator?.givenName, curator?.familyName))}</StyledAvatar>
```

Reply:
Men klarer som sagt ikke gi noen god grunn til at noen av de løsningene er bedre, for jeg tror ikke sånn det er nå vil trigge noen flere rerenders enn tilfellet er akkurat nå heller 😅 Bare gjør det du selv syns virker mest naturlig av de 3 mulige løsningene :)

Reply:
Enig i at dette burde være en egen komponent i en egen fil. Men stylingen (størrelsen) er manuelt tilpasset en select-komponent med size='small'. Kunne man styrt dette på elegant vis i komponenten? Denne komponenten kommer sikkert til å vokse og gjenbrukes flere plasser (og da med bilde mest sannsynlig). 
Mulig jeg tenker for langt fram i tid da, og at dette er ting som kan tas når det trengs.

Reply:
Vi bruker vel i utgangspunktet kun to størrelser på inputs, `"small"` eller default. Kunne du lagt til en prop op komponenten, `isSmall`, og så brukt en ternary får å definere størrelsen? `sx={{ scale: {isSmall ? 0.8 : 1 } }}`? Eller misforstod jeg problemet? 😅

Reply:
Liker idèen! Men om man bruker komponenten en annen plass i løsningen (utenfor en input-komponent), hvilket forhold skal man ha til `isSmall` da? Kunne man heller videreført `size`-propertien til `TextField`, og heller lagt til `inputAdornment` som size option? Slik at man får `<"small" | "medium" | "inputAdornment">` som valg for size?

Reply:
Oppsummering etter diskusjon IRL: Avatar brukes flere steder i løsningen, så det må uansett tas en oppvask og få samlet disse. Tar denne også når den tid kommer.

Reply:
Jeg satt <ProfilePicture> inni Mui ListItemAvatar.
https://github.com/BIBSYSDEV/NVA-Frontend/blob/develop/src/pages/basic_data/app_admin/UserList.tsx